src/buffer.rs

```diff
@@ -40,6 +40,7 @@ pub(crate) struct Buffer {
     decorations: Vec<Decoration>,
     selection_set_history: History<SelectionSet>,
     dirty: bool,
+    user: bool,
```

Comment:
Let's add a documentation comment to explain the necessity of this field. I don't want to solely rely on PR comments as that will be hard to trace if this field is moved to another struct, or if this struct is moved to another file.

Reply:
I did have a comment on it initially, but I then decided that it was private and never really used by someone outside of this file. Therefore, I added a comment to the `pub fn user()` function which describes what it is used for. That will show up in the editor where people are using it.

I'd be happy to add a comment to the field as well.

Reply:
I think you can move the comments from the method to the field, and then document the method saying something like:
```rs
/// Refer `Buffer::opened_by`
```